sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -264,38 +264,149 @@ abstract class GpuExplodeBase extends GpuUnevaluableUnaryExpression with GpuGene
       }
   }
 
+  private def getPerRowRepetition(explodingColumn: ColumnVector, outer: Boolean): ColumnVector = {
+    withResource(explodingColumn.countElements()) { arrayElements =>
+      if (outer) {
+        // for outer, empty arrays and null arrays will produce a row
+        withResource(GpuScalar.from(1, IntegerType)) { one =>
+          val noNulls = withResource(arrayElements.isNotNull) { isLhsNotNull =>
+            isLhsNotNull.ifElse(arrayElements, one)
+          }
+          withResource(noNulls.greaterOrEqualTo(one)) { isGeOne =>
+            isGeOne.ifElse(noNulls, one)
+          }
+        }
+      } else {
+        // ensure no nulls in this output
+        withResource(GpuScalar.from(0, IntegerType)) { zero =>
+          GpuNvl(arrayElements, zero)
+        }
+      }
+    }
+  }
+
+  private def getRowByteCount(column: Seq[ColumnVector]): ColumnVector = {
+    val repeatTable = new Table(column:_*)
+    val bits = withResource(repeatTable) {
+      _.rowBitCount()
+    }
```

Comment:
How about this?

```suggestion
    val bits = withResource(new Table(column:_*)) { repeatTable =>
      repeatTable.rowBitCount()
    }
```

Reply:
better? https://github.com/NVIDIA/spark-rapids/pull/6631/commits/19ea18a3047adf6bb7133948905c5c91db6b7912